### DISCUSSION TITLE
The 6.0.0 icons are a big change

### DISCUSSION THREAD

**Discussion by Sunspark-007** [SylEleuth/gruvbox-plus-icon-pack#112] [General]
I'm a bit on the fence about some of the design decisions.. I'll give two examples.. the monochrome startbar icon for Dolphin is a computer monitor with 3 squares on it.. but Dolphin is a file explorer.. a folder is a more intuitive shape I feel. Kwrite is two quotation marks.. wouldn't a rectangle with a pen be more intuitive? Etc.

Another issue is the colour icons in favourites, etc. are now squares with rounded corners a'la Samsung except.. a bunch of them are now LIGHT.. so on a dark theme there is a lot more white now than there was before.. the older icons (maybe they were the original app icons) were larger and had less white in them.. I took a look and while alt icons exist for some, it seems hard to change them.. for example:
![image](https://github.com/user-attachments/assets/907ac2db-1f69-4b48-8a75-f59c6d5280d5)
But if you try to change it on an immutable system like this one, let's say to the dark alt, it says "Could not save properties due to insufficient write access to: path". I'm a little puzzled to be honest, because shouldn't settings like this just be stored on my home folder which does have full write access? So the yuzu icon stays white.

**Comment by SylEleuth** [comment]
Please attach screenshot to every example You presented because in most parts I didn't get what You tried to tell me.

What I can say now:
- there are no `Yuzu-alt` icons in that pack, they were removed, which means I don't know what version You are using (that is why Your system couldn't find a path to it)
- what is `monochrome startbar` in that example?
- all icons for `dolphin` are represented by folder icon, for `kwrite` it's a pen and paper
- where are Your `favourites`?
- some colourful icons have a bright background (it's not white though), some icons might not be great for bright screen (e.g. wallpaper) but that's why they have darker border and that should give a slightly better visibility; icons without that border will get it in the future.

PS. Because of a mistake I've made I've updated new zip file for the latest release. It's not directly related to Your problems though.

**Comment by Sunspark-007** [comment]
The monochrome startbar is this, I'm guessing the name is Symbolics. From top to bottom, start button, DOLPHIN file explorer, calculator, KWRITE, Firefox, System Monitor, System Settings.. so, the ones in all caps are the ones you said should be different icons.. So I figured out what you see.. in order to get the folder icon and the notepad icon which are colour icons, I have to make that start bar 70 pixels wide. That's way too much on a non-4K display. So yeah, two issues now.. symbolics icons need a second looking at, and the scaling for the colour icons might want to be considered to be allowed for sizes lower than 70 pixels.
![image](https://github.com/user-attachments/assets/47c33633-be70-44f4-8bcf-b488d00a6c05)

And here is what the icons look like with the gruvbox-plus-LIGHT theme.. Looks like system monitor is different too. The colour icon for it in the favourites menu is H-shaped like the dark one but with the light icon theme it's something else..

![image](https://github.com/user-attachments/assets/38d3c70c-cd2f-4d35-ba5d-a84dc079a8ff)

The version I was using was 5.5, then 6.0 updated through discover store, and now 6.0.1 also updated through discover store.

Favourites is when you click on the start button and it's all the quick launch icons on the topmost menu item. This is with KDE Plasma.

I was poking around last night, the .desktop files seem to be owned by root root, so that's probably why I couldn't try manually changing the icon to one of the others that you say don't exist anymore. I will need to figure out permissions later, but I am puzzled why you say they don't exist. If I look at /Home/.local/share/icons/Gruvbox-Plus-Dark they're right there.. 
![image](https://github.com/user-attachments/assets/cb63abcd-5539-4223-a39a-00398bed1f07)

The graduations on some of the larger icons are not ideal.. like here is VLC's.. should I really be able to see the vertical line running right through it? Why not dither them? Heroic's icon has a horizontal line.
![image](https://github.com/user-attachments/assets/5821c734-7fbc-41a7-ac21-52884199523b)
![image](https://github.com/user-attachments/assets/52abfa25-99b1-4eb3-a0c3-17f09453669c)

**Reply by Sunspark-007** [reply]
PS. Please take these comments as constructive feedback and not criticism. I do like this project.

**Comment by SylEleuth** [comment]
I will spend more time on it tomorrow but right now I see many issues here and not on my side.

![Screenshot_2024-12-08-120225](https://github.com/user-attachments/assets/35b899fb-ef1c-4e7a-a429-3525814c7897)
1. This is search for `yuzu` in `apps/scalable` from the latest release (and git, and anywhere on my computer).
2. Your icon for system monitor is for `htop` app. It's exactly the same as in light folder.
3. First icon (start button) doesn't exist in that repo.
4. Scaling for colourful icons starts from 32px .
5. How looked `kwrite` icon in 5.5 version?
6. I need to see those `favourites` of Yours.
7. If Your .desktop files You want to change are in `/usr/share/applications` don't touch it. Copy to `~/.locale/share/applications` and change it there.
8. Some `scalable` icons have a transparent horizontal or vertical bars (or both). It's a feature not a bug.
9. I need all details about Your system and DE.

I've just installed it on a fresh arch kde and everything is working as it should. My suggestion on the start is uninstalling the pack completely (if there is any trace of it in `~/.locale/share/icons,` delete it too) and installing again.

**Comment by Sunspark-007** [comment]
Thanks for the suggestion. So the Windows approach (lol) did work here.

I changed the theme, uninstalled Gruvbox Plus using discover app, then went into the icons folder to see if the folders were still there. They were gone. Had a momentary scare when some other icons started disappearing, but they eventually repopulated after awhile.

Then I went into appearance>icons>get new so to ensure it would install through that process. Tried the dark theme, and the symbolic icons are what they are supposed to be. Also interesting, looked at the full size ones and the system monitor icon is now the heartbeat line and not the H-shaped one that it was before.

Took a quick look, and now I only have the 4 yuzu icons that are named what you had.

Something must have been really glitched, but I don't understand what. Thanks for helping. :)

Going through the app launcher menus.. I see some symbolic icons.. Like, in the Office subcategory, I have colour icons for 4 apps, but the libreoffice flatpak is using symbolic icons instead of the colour ones that I verified are in the svg folder..

E.g.
![image](https://github.com/user-attachments/assets/f9fc6585-a275-4486-a2a5-5b925bafecf7)

I scrolled through "All Applications" and it seems to be only LibreOffice and Audacious using the symbolic icon. Is that a bug, or should I try something to resolve that?

PS. What about the panel? Should these icons be bolder like the peek at desktop one at the very bottom?

![image](https://github.com/user-attachments/assets/c0cf2f19-ec4b-4a92-b886-e2ed9c1338cc)

Same with this.. Places icon seems bolder but the others are not.

![image](https://github.com/user-attachments/assets/58c0087e-e862-41b0-bdae-74338d9da7fa)

Operating System: SteamOS 3.6.20 Build: 20241030.1
KDE Plasma Version: 5.27.10
KDE Frameworks Version: 5.115.0
Qt Version: 5.15.12
Kernel Version: 6.5.0-valve22-1-neptune-65-g9a338ed8a75e (64-bit)
Graphics Platform: X11

**Comment by SylEleuth** [comment]
I'm glad that it worked. I had a suspicion that something messed up during installation process.

I don't know what is going on with `Writer` and `Audacious`. Monochrome icons are exactly the same as as scalable (same names) so they should be used depending on the size. Can you show me Your .desktop file for them? Also flatpak (and snap) have issues with properly displaying icons but I don't about that case.

Panel is a different matter. Starting version 6 plasma started using icons from the pack in the panel. But before that most important icons were bundled in plasma style. I'm using version 5.27 myself and had to create gruvbox style. You can check it [here](https://github.com/SylEleuth/gruvbox-plus-kde). Plasma style is located in `plasma/desktoptheme`. Link it or copy to `~/.locale/share/plasma/desktoptheme/`. If you don't like colours You would have to modify `colors` file (not an easy task).

**Comment by Sunspark-007** [comment]
Right-clicking in the menu and selecting "Edit Application" to see the path states /var/lib/flatpak/app/org.libreoffice.LibreOffice/current/active/export/share/applications/org.libreoffice.LibreOffice.writer.desktop

Audacious is /var/lib/flatpak/app/org.atheme.audacious/current/active/export/share/applications/org.atheme.audacious.desktop

[Two desktop files.zip](https://github.com/user-attachments/files/18067809/Two.desktop.files.zip)

Desktop files for the rest of the libreoffice suite will be the same as writer.

Btw, at the smallest zoom level in a folder /usr/share/applications/ I don't spend time in, I noticed this:
![image](https://github.com/user-attachments/assets/d012a947-5891-4bb7-8af7-8b7bf04ca0a9)
![image](https://github.com/user-attachments/assets/da283a88-8ac2-4fa4-a729-05831487257d)

Everything in that folder is symbolic except for those 4 in colour.

Why don't you have that Plasma Desktop theme in kde's get new plasma styles? It's not there, so to manually install it what I did was use a github extractor webpage to download the plasma directory from github so I could get the folders as they are. Pasted the two theme folders into /home/deck/.local/share/plasma/desktoptheme/ went back to system settings>appearance>plasma to see if they were there.. they were! Trying it out.. found two small issues.. entering the edit mode on the start bar where it has panel width, there is an icon for + but the icon for - has disappeared. Steam icon I think needs to be part of the theme, it doesn't currently have an icon in the folder.

Places icon also still looks off in size proportion.
![image](https://github.com/user-attachments/assets/f464a619-54eb-49f9-afd1-08e9b9c11aa4)

Dark theme has a single pixel border around it.. distracting.. is this by design?
![image](https://github.com/user-attachments/assets/4bbd46c2-9dc2-4597-864d-436a58088824)

PS. I'm going to suggest in the launcher that you don't use the same gear icon for both Settings and System. Probably an icon of a computer monitor is fine for System.
![image](https://github.com/user-attachments/assets/1a80aa19-8906-410a-b381-dd876c38007e)

**Comment by SylEleuth** [comment]
I don't know much about flatpak but I've already mentioned that it has issues with icons. Do You see similar `.desktop` files in `/usr/share/applications`? If not then icons are already there and it's flatpak's fault. And please, don't send/post zip files. Copy content of the `.desktop` file and paste it here, or at least a line starting with `icon=`,

> Btw, at the smallest zoom level in a folder /usr/share/applications/ I don't spend time in, I noticed this:

Zoom where? How it looks at the largest zoom? You have to show me the content of that `.desktop` files so I could see what icon is used.

> Why don't you have that Plasma Desktop theme in kde's get new plasma styles

Because it's loo late to publish it anywhere (except github). Plasma is in version 6.2 now and the only people using 5.27 version are You and I.

> Steam icon I think needs to be part of the theme, it doesn't currently have an icon in the folder.

I don't understand, what folder?

> Places icon also still looks off in size proportion.

Application Launcher takes `Places` icon from `actions`, probably 22px size, and the rest of the icons are 16px.

> Dark theme has a single pixel border around it.. distracting.. is this by design?

By design. Probably I (or You) could change it but it would require some time and effort. I have no plans in touching it except when I'll move to Plasma 6.

> gear icon for both Settings and System

There is an application called `System Settings`. What are those in Your case?

PS. I need more (detailed) information every time You ask for something like that.

**Reply by Sunspark-007** [reply]
> Copy content of the .desktop file and paste it here, or at least a line starting with icon=,

Desktop entry is: Icon=org.libreoffice.LibreOffice.writer
Desktop action new document is: Icon=document-new

Icon=org.atheme.audacious

> Zoom where? How it looks at the largest zoom? You have to show me the content of that .desktop files so I could see what icon is used.

The slider in the bottom right. Ok, at 16 & 22 pixels, only those 4 are in colour. At 32 pixels all are in colour except for 10 symbolic. 48 pixels 12 symbolic, 64 and up pixels 13 symbolic. Looking at them, they mostly seem like handlers and stuff. 

E.g. Icon=applications-system-symbolic for Exec=/usr/lib/xdg-desktop-portal-gtk

So for the 16 pixel size that are in colour,
Name=Install Firefox
Icon=/usr/share/icons/hicolor/scalable/apps/install-firefox.svg

Example for one of the blue ones,
Name=IBus Pinyin Setup
Exec=/usr/lib/ibus/ibus-setup-pinyin
Icon=/usr/share/ibus-pinyin/icons/ibus-pinyin.svg

> I don't understand, what folder?

I'm saying here, for the plasma desktop theme, that you should think about making a symbolic Steam icon since there are a vast number of installations with it. One doesn't currently exist.

> There is an application called System Settings. What are those in Your case?

The issue isn't with the applications here, but the menu categories. I can't figure out where KDE stores the path for the menu categories. Do you know?

With Breeze Dark, this is how they are differentiated.
![image](https://github.com/user-attachments/assets/a30908c8-ef6f-47ef-9dc5-99e1f7e3f5de)
And another theme.
![image](https://github.com/user-attachments/assets/6f172f26-b100-4205-bff7-b415ca2c70fc)

Those do populate a colour icon for libreoffice and audacious in the kde launcher menu, then again they don't also have a symbolic pack with competing rules.

**Comment by SylEleuth** [comment]
I've added all missing `libreoffice` and `audacious` icons.
`applications-system-symbolic` - symbolic icons are... symbolic, they don't have scalable counterpart.

> The issue isn't with the applications here, but the menu categories

Menu category in where, launcher menu? What launcher? I don't use any so You have to give me all details.

**Reply by Sunspark-007** [reply]
The KDE application launcher.

![image](https://github.com/user-attachments/assets/03e435ce-db8b-4e07-963c-cbc30ff04813)

**Comment by SylEleuth** [comment]
Whatever launcher it is (I still don't know) I can't even test it right now. But I've made some changes to the settings icons, so You can check it out. https://github.com/SylEleuth/gruvbox-plus-icon-pack/commit/ba8f9a8ed8144749a37ea5763d75fa0394fd277d

**Reply by Sunspark-007** [reply]
What I find confusing when you say you don't know what launcher it is, you said you were using Plasma 5.27.. shouldn't you just know automatically what it is because everyone has the same one?

Or are you using a third party launcher?

It's called the Kickoff Application Launcher

![image](https://github.com/user-attachments/assets/8c4f731e-454e-4a1b-86b8-13f0c7b14e59)

**Reply by Sunspark-007** [reply]
I've made some progress on figuring it out.. if I right click on the launcher icon and select edit applications I can see the settings and system icon.. with the current release version the two are grey and orange versions at larger sizes.. but the launcher is using the symbolic icon.. The new design you made is nice, but should there be a symbolic icon as well? 

preferences-system-symbolic
applications-system-symbolic

I'm thinking applications-system could be one of the existing like
![image](https://github.com/user-attachments/assets/e9878d32-1d2a-40a4-95ec-9490efd9457d) or perhaps fill in the hollow center so it is a solid gear.
![image](https://github.com/user-attachments/assets/30002c3f-9bb7-496e-a239-b44c213ea74e)


preferences-system-symbolic could be
![image](https://github.com/user-attachments/assets/feceff36-d298-49a6-b72d-206e3a401f66)

It does look to me though, that you already have a second gear icon.. one with a wrench.. so settings menu/preferences should have that one.

**Comment by SylEleuth** [comment]
"Application launcher" is a generic name and as You can see Yourself You were talking about `Kickoff`. I'm not using launcher myself, only `KRunner` and `Latte`, but years ago I tried many different ones and didn't uninstall them. Also don't know which is which because they are not working (it's related to broken dynamic library). I'm on Arch and when KDE went to Plasma 6 I've downgraded and blocked Plasma from upgrading which caused some issues (partial upgrade is not recommended). Now I can't run launchers. It's not a problem while I'm not using them, but also I can't test it. And also my VM broke recently and can't test it on latest Plasma so I have to figure it out soon.

If You can't test changes I've made nothing You will tell me will be helpful. You would have to wait until I figure it out myself.

**Reply by Sunspark-007** [reply]
I'm curious.. why do you want to stay on 5.27? In my case, it's because my device with KDE is on Valve Time and it won't get updated until they feel like it. Their mainline does have 6, but that could be a year away.

Boxes VM is in flathub, you could try it out using that install method without affecting the rest of the system.

**Reply by SylEleuth** [reply]
I'm extremely attached to several widget that don't work properly or not exist any more in Plasma 6. But most important thing is tiling manager. I have 2 (3) monitors and using tiling on version 6 is a nightmare. Linux is not usable for me without tiling and if my Plasma 5 break down somehow to the point I won't be able to use I'd rather move to i3wm.

I'm testing icons on Plasma 6 installed on VM so I'm regularly on newest version and sadly it's still not where I want it to be.

**Reply by kostiato** [reply]
offtopic: 
please give a try in VM also to https://github.com/anametologin/krohnkite Plasma 6 fork
for me it is working very smooth and pretty stable in Plasma 6
also there are [upcoming QoL changes](https://blogs.kde.org/2024/12/07/this-week-in-plasma-oodles-of-features/#notable-ui-improvements) for using native Quick Tiling in Plasma 6.3.0

**Reply by SylEleuth** [reply]
`Krohnkite` it's the only viable option. I've been using it for a long time, and now using `Bismuth`. `Krohnkite` is not working with two monitors in Plasma 6 well. And have some other problems. Also I can't turn off virtual desktops on the other monitor but it's Plasma's fault.

**Comment by SylEleuth** [comment]
Repaired VM, found proper icons, changed `system` icons https://github.com/SylEleuth/gruvbox-plus-icon-pack/commit/967aae2b7fc0d5ead41e6dfe25e880f6c22465a3

**Reply by Sunspark-007** [reply]
Hi, I have the latest update (via flatpak).. the settings and system icons in the kde application launcher are still the same cog icon symbolic icon instead of two different cog icons.

When I look at the preview in the edit applications list one is gray and the other is orange.

I think the symbolic icon should be different though.. a cog with a wrench in it exists so I think it should be used.

Is there additional information I can provide you that will help, or you have enough info to figure this out?

**Reply by SylEleuth** [reply]
I need screenshot for everything what You said.

**Reply by Sunspark-007** [reply]
How it looks in the kickoff launcher:
![image](https://github.com/user-attachments/assets/7c103737-7b1d-427c-b1de-25fca97b9f95)

How it looks in the edit applications list for the kickoff launcher:
![image](https://github.com/user-attachments/assets/b15ef4bb-5442-4eb9-bb23-312bf43d1893)

![image](https://github.com/user-attachments/assets/23adbadb-3cd1-44c0-bc6e-a56d48bb76c7)
![image](https://github.com/user-attachments/assets/a083534e-5c68-45a2-9940-24cd2227cd6b)

I'm saying that one of the two symbolic gears should be the one with the wrench instead of both the same. Likely the settings one.
![image](https://github.com/user-attachments/assets/657bb173-4ebb-4462-8064-d4fd31b0548e)

**Comment by SylEleuth** [comment]
Sorry, but I've moved to Plasma 6. I will have to search for distro with the old version. It might take time.

You mention that You installed it from Flatpak. How so?

**Reply by Sunspark-007** [reply]
That was before. You said last time not to do the update through discover even if it shows up there (I imagine it only really works for point updates) so I uninstalled the whole thing and re-installed it by adding it through the dialogs.

3.7 of SteamOS will be on Plasma 6.. I'm guessing they will put that out in June.

I don't think it is a Plasma thing though.. I think it is pointing to the same icon for both entries.

**Reply by SylEleuth** [reply]
Can You point where I said to not update it through discover? I can't find it.

On Plasma 6 there is only `System` in launcher, there is no `Settings`, as a main category. And because every other icon pack uses the same clog icon for `System` I've decided to not change it. Please wait for version 6 and see then how it works for You.
